README.md

```diff
@@ -54,248 +99,292 @@ $(SRCROOT)/Carthage/Build/iOS/CryptoSwift.framework
-     } catch let ostError {
+	* Application must confirm to `OstWorkflowDelegate` protocol. The `OstWorkflowDelegate` protocol defines methods that allow application to interact with `OstWalletSdk`.
 
-     }
-     return true
-}
+
+2. `Getters`: These functions are synchronous and will return the value when requested.
+
+
+
+## Workflows
+
+### 1. initialize
+
+You must initialize the SDK before start using it.
+
+Recommended location to call **OstWalletSdk.initialize()** is in [application](https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1622921-application) method of [UIApplicationDelegate](https://developer.apple.com/documentation/uikit/uiapplicationdelegate). 
+
+```
+OstWalletSdk.initialize(apiEndPoint: String)
 ```
 
-### Set up the device
+| Parameter | Description |
+|---|---|
+| **apiEndPoint** <br> **String**	| OST PLATFORM API ENDPOINT: <br> 1. Sandbox Environment: `https://api.ost.com/testnet/v2/` <br> 2. Production Environment: `https://api.ost.com/mainnet/v2/` |
 
-The `setupDevice` API should be called after user login or signup is successful.
 
-Once the user is logged in, then `setupDevice` should be called every time the app launches, this ensures that the current device is registered before communicating with OST Platform server.<br/><br/>
-**Parameters**<br/>
-&nbsp;_userId: OST Platform user id provided by application server_<br/>
-&nbsp;_tokenId: Token id provided by application server_<br/>
-&nbsp;_delegate: Callback implementation object for application communication_<br/>
 
-```Swift
+<br>
+### 2. setupDevice
+This workflow needs `userId` and `tokenId` so `setupDevice` should be called after your app login or signup is successful.
+Using the mapping between userId in OST Platform and your app user, you have access to `userId` and `tokenId`.
+
+**If the user is logged in, then `setupDevice` should be called every time the app launches, this ensures that the current device is registered before communicating with OST Platform server.**
+
+
+```
 OstWalletSdk.setupDevice(
     userId: String,
     tokenId: String,
     delegate: OstWorkflowDelegate
-    )
-```    
+)
+```
+
+| Parameter | Description |
+|---|---|
+| **userId** <br> **String**	| Unique identifier of the user stored in OST Platform  |
+| **tokenId** <br> **String**	| Unique identifier of the token economy |
+| **delegate** <br> **OstWorkflowDelegate**	|An object that implements the callback function available in `OstWorkflowDelegate` protocol. These callback functions are needed for communication between app and wallet SDK. Implement `flowComplete` and `flowInterrupt` callback functions to get the workflow status. Details about other callback function can be found in [OstWorkflowDelegate protocol reference](/platform/docs/sdk/references/wallet_sdk/iOS/latest/protocols/).<br> This object should implement `registerDevice` callback function. `registerDevice` will be called during the execution of this workflow.  |
```

Comment:
object dont implement delegates. Classes do. So, it should be:
instance of a class that implements the delegate/protocol.

Reply:
The link to `OstWorkflowDelegate` is also broken. Please fix it.